api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -304,6 +305,53 @@
    * @since 0.1.0
    */
   interface Builder {
+    /**
+     * Sets the parent {@code Span} to use. If not set, the value of {@code Tracer.getCurrentSpan()}
+     * at {@link #startSpan()} time will be used as parent.
+     *
+     * <p>This <b>must</b> be used to create a {@code Span} when manual Context propagation is used
+     * OR when creating a root {@code Span} with a parent with an invalid {@link SpanContext}.
+     *
+     * <p>Observe this is the preferred method when the parent is a {@code Span} created within the
+     * process. Using its {@code SpanContext} as parent remains as a valid, albeit inefficient,
+     * operation.
+     *
+     * <p>If called multiple times, only the last specified value will be used. It is an error to
+     * specify a {@code null} parent.
+     *
+     * @param parent the {@code Span} used as parent.
+     * @return this.
+     * @since 0.1.0
+     */
+    Builder setParent(Span parent);
+
+    /**
+     * Sets the parent {@link SpanContext} to use. If not set, the value of {@code
+     * Tracer.getCurrentSpan()} at {@link #startSpan()} time will be used as parent.
+     *
+     * <p>Similar to {@link #setParent(Span parent)} but this <b>must</b> be used to create a {@code
+     * Span} when the parent is in a different process. This is only intended for use by RPC systems
+     * or similar.
+     *
+     * <p>If no {@link SpanContext} is available, users must call {@link #setNoParent()} in order to
```

Comment:
You should document that `null` is invalid - based on your noop implementation.

Reply:
Well, that's why I wanted to keep a few things 'repeated' in this and the other method's docs - it might seem too verbose to have many things repeated, but given that semantically they are not the same, this could be handy.

So I I will also restore that the last specified value will be used.